test/ft/constexpr.cpp

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright (c) 2016-2018 Kris Jusiak (kris at jusiak dot net)
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+#include <boost/sml.hpp>
+#include <string>
+#include <typeindex>
+#include <typeinfo>
+#include <utility>
+#include <vector>
+
+namespace sml = boost::sml;
+
+struct e1 {};
+struct e2 {};
+
+
+constexpr auto idle = sml::state<class idle>;
+constexpr auto s1 = sml::front::state<class s1>{};
+
+using namespace sml;
+#if !defined(_MSC_VER)
+constexpr auto s2 = "s2"_s;
+constexpr auto event1 = "e1"_e;
+#else
+constexpr auto s2 = event<e2>;
```

Comment:
This guy should probably be `s2 = state<class s2>{}`

Reply:
right, will fix